src/processor/tree/tree_wrapper.rs

```diff
@@ -79,35 +96,68 @@ impl TreeWrapper {
             hex::encode(root_hash)
         );
 
-        assert_eq!(root_hash.as_bytes(), block.new_state_root);
-        tracing::debug!("Successfully processed block {}", block.l2_block_number);
+        let root_hash_bytes = root_hash.as_bytes();
+        if root_hash_bytes == block.new_state_root {
+            tracing::debug!("Successfully processed block {}", block.l2_block_number);
+
+            Ok(root_hash)
+        } else {
+            tracing::error!(
+                "Root hash mismatch: {:?} vs. {:?}",
```

Comment:
This might make it more clear, which is which:
```suggestion
                "Root hash mismatch!\nLocal: {:?}\nPublished: {:?}",
```